src/evidently/metrics/data_integrity/column_summary_metric.py

```diff
@@ -435,38 +427,41 @@ def get_text_stats(
 
 @default_renderer(wrap_type=ColumnSummaryMetric)
 class ColumnSummaryMetricRenderer(MetricRenderer):
-    def render_json(self, obj: ColumnSummaryMetric) -> dict:
-        result = dataclasses.asdict(obj.get_result())
-        result.pop("plot_data", None)
-        return result
-
     def render_html(self, obj: ColumnSummaryMetric) -> List[BaseWidgetInfo]:
         metric_result = obj.get_result()
         column_type = metric_result.column_type
         column_name = metric_result.column_name
         # main plot
-        bins_for_hist = metric_result.plot_data.bins_for_hist
+        bins_for_hist: Histogram = metric_result.plot_data.bins_for_hist
         if bins_for_hist is not None:
-            hist_curr = bins_for_hist["current"]
+            hist_curr = bins_for_hist.current
             hist_ref = None
             metrics_values_headers = [""]
-            if "reference" in bins_for_hist.keys():
-                hist_ref = bins_for_hist["reference"]
+            if bins_for_hist.reference is not None:
+                hist_ref = bins_for_hist.reference
                 metrics_values_headers = ["current", "reference"]
 
             if column_type == "cat":
-                fig = plot_distr(hist_curr=hist_curr, hist_ref=hist_ref, color_options=self.color_options)
+                fig = plot_distr(
+                    hist_curr=hist_curr,
+                    hist_ref=hist_ref,
+                    color_options=self.color_options,
+                )
                 fig.update_layout(legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="right", x=1))
                 fig = json.loads(fig.to_json())
             if column_type == "num":
-                ref_log = None
-                if "reference_log" in bins_for_hist.keys():
-                    ref_log = bins_for_hist["reference_log"]
+                # always should be present for num columns
+                assert bins_for_hist.current_log is not None
```

Comment:
I think it would be better to use `if ... raise ValueError()` with some meaningful message so it would b easier to understand if it fails.